### ISSUE TITLE
Dependency error with marked

### ISSUE THREAD

**Issue by brichet** [jupyterlab/jupyterlab#15020]
## Description

A fresh install of the npm packages resolves with `marked=7.0.4` (when generating again the `yarn.lock` file)
```
├─ @jupyterlab/markedparser-extension@workspace:packages/markedparser-extension
│  └─ marked@npm:7.0.4 (via npm:^7.0.2)
```

It seems this patch release break the API, and the line
https://github.com/jupyterlab/jupyterlab/blob/ac8f9e2f47b271785127757541a6fd8aa10c4d31/packages/markedparser-extension/src/index.ts#L154
cannot be compiled:
```
Type '(token: Token) => Promise<void>' is not assignable to type '(token: Token) => void | unknown[]'.
  Type 'Promise<void>' is not assignable to type 'void | unknown[]'.ts(2322)
marked.d.ts(606, 9): The expected type comes from property 'walkTokens' which is declared here on type 'MarkedOptions'
```
## Reproduce

Install the main branch of jupyterlab will download the latest release of `marked`.

## Context

Adding a resolution to the main `package.json` file fixes the issue, but it should probably be fixed in jupyterlab.
```
  "resolutions": {
    "@jupyterlab/services/ws": "8.12.0",
    "@types/react": "^18.0.26",
    "marked": "7.0.2",
    "react": "^18.2.0",
    "yjs": "^13.5.40"
  },
```

**Comment by krassowski** [comment]
The problematic `unknown` was removed in https://github.com/markedjs/marked/pull/2955 which was included in [v7.0.5](https://github.com/markedjs/marked/releases/tag/v7.0.5) release.

Does it works all good now?

**Comment by brichet** [comment]
Yes, it's fixed now.
Thanks @krassowski for taking care of it.
